superbench/benchmarks/micro_benchmarks/directx_utils/D3D12Timer.h

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+#include <d3d12.h>
+
+namespace D3D12 {
+struct GPUTimestampPair {
+    UINT64 Start;
+    UINT64 Stop;
+};
+
+enum QueueType { compute = 0, copy = 1 };
+
+// D3D12 timer.
+class D3D12Timer {
+  public:
+    // Constructor.
+    D3D12Timer();
+
+    // Destructor.
+    ~D3D12Timer();
+
+    void init(ID3D12Device *pDevice, ID3D12CommandQueue *pCommandQueue, UINT numTimers, QueueType type);
+
+    // Start timestamp.
+    void start(ID3D12GraphicsCommandList *pCommandList, UINT timestampPairIndex);
+
+    // Stop timestamp.
+    void stop(ID3D12GraphicsCommandList *pCommandList, UINT timestampPairIndex);
+
+    // Resolve query data. Write query to device memory. Make sure to wait for query to finsih before resolving data.
+    void resolveQueryToCPU(ID3D12GraphicsCommandList *pCommandList, UINT timestampPairIndex);
+
+    // Get start and end timestamp pair.
+    double getElapsedMsByTimestampPair(UINT timestampPairIndex);
+
+    // Get the GPU frequency.
+    double getGPUFrequecy() { return m_gpuFreqInv; }
+
+  private:
+    ID3D12Device *m_device = nullptr;
+    ID3D12QueryHeap *m_queryHeap = nullptr;
+    ID3D12Resource *m_queryResourceCPU = nullptr;
+    bool m_active = false;
```

Comment:
>ool m_active = false; [](http://example.com/codeflow?start=5&length=21)

since we don't use this variable, so I suggest we can remove it.

Reply:
removed